### PR TITLE
Refactored the testWorkflowForkApp test case to make it more deterministic.

### PR DESCRIPTION
**Pull request by sagarkapare**

### CODE REVIEW

File: cdap-app-fabric/src/test/java/co/cask/cdap/WorkflowAppWithFork.java [hunk 1]
```diff
@@ -17,87 +17,75 @@
...
-import java.util.Map;
+
+import java.io.File;
+import java.util.concurrent.TimeUnit;
 
 /**
  *
  */
 public class WorkflowAppWithFork extends AbstractApplication {
+  public static final String SYNCH_ON_FILE = "/tmp/WorkflowAppWithFork.done";
```
Comment (chtyim) [inline cdap-app-fabric/src/test/java/co/cask/cdap/WorkflowAppWithFork.java]: Don't write to tmp. Takes a runtime arg and have the unit-test create a temp file (using `TemporaryFolder`) and pass in the pass through runtime arg.

Comment (sagarkapare) [inline cdap-app-fabric/src/test/java/co/cask/cdap/WorkflowAppWithFork.java]: Using `TemporaryFolder` now.

**Comment by sagarkapare** [comment]
@chtyim Addressed the comments. Please review.

File: cdap-app-fabric/src/test/java/co/cask/cdap/WorkflowAppWithFork.java [hunk 2]
```diff
@@ -17,87 +17,90 @@
...
+  static class OneAction extends AbstractWorkflowAction {
+    private static final Logger LOG = LoggerFactory.getLogger(OneAction.class);
+    private String doneDir = "";
 
     @Override
-    public void configure() {
-      setName("AnotherWordCountMapReduce");
-      setDescription("WordCount job from Hadoop examples");
+    public void initialize(WorkflowContext context) throws Exception {
+      doneDir = context.getRuntimeArguments().get("done.directory");
```
Comment (chtyim) [inline cdap-app-fabric/src/test/java/co/cask/cdap/WorkflowAppWithFork.java]: If you have the argument as `done.file`, then you don't even need the `SYNCH_ON_FILE` constant.

**Comment by chtyim** [comment]
Just one comment left. Otherwise LGTM.

File: cdap-app-fabric/src/test/java/co/cask/cdap/WorkflowAppWithFork.java (hunk 2 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/test/java/co/cask/cdap/WorkflowAppWithFork.java]: Change the argument to pass the file name itself.

**Comment by sreevatsanraman** [comment]
Build: http://builds.cask.co/browse/CDAP-DUT1201

**Comment by sreevatsanraman** [comment]
LGTM :+1: Test pass. Merging

**Pull request merged**